.conda/env_dev.yml

```diff
@@ -0,0 +1,25 @@
+name: um2nc-dev
+
+channels:
+    - conda-forge
+    - accessnri
+    - coecms
+    - default
+    - nodefaults
+
+dependencies:
+    - python >=3.10,<=3.12
+    - cf-units
+    - cftime
+    - f90nml
+    - ipykernel
+    - mule
+    - netcdf4
+    - numpy
+    - pip
+    - pytest
+    - pytest-cov
+    - pylint #TODO: remove this after setting up linting with ruff
+    - ruff
```

Comment:
Are we in a position to definitively cast `pylint` aside in favour of `ruff`? My understanding (from a very fast and superficial Google search) is that `ruff` is a lot faster, and is seen as the 'sexier' option, but `pylint` still gives more complete coverage.

Given this is the development dependency list anyway, I'm wondering if leaving both options available might keep more people happy.

Reply:
I am a huge fan of Ruff, and [this comment](https://github.com/ACCESS-NRI/um2nc-standalone/issues/26#issuecomment-2398034872) in #26 sets the plan for the setup of ruff pre-commit linting/formatting.

I think it will be the next step after this PR. 
For now we will leave it like this and change it when there it the PR about pre-commit hooks.

I am ok leaving `pylint` there (even though I am not sure what use that will do).